Movie Search/ViewController.swift

```diff
@@ -0,0 +1,121 @@
+    var genreMap: [IndexPath: [Genre]] = [:]
+    
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        let bundle = Bundle(for: type(of: self))
+        let nib = UINib(nibName: "MovieCollectionViewCell", bundle: bundle)
+        movieCollectionView.register(nib, forCellWithReuseIdentifier: "MovieCollectionViewCell")
+    
+    }
+    
+    @IBAction func searchButtonTapped(_ sender: Any) {
+        DispatchQueue.main.async (qos: .default) {
+            self.getMoviesNowPlaying()
+        }
+    }
+}
+
+//Data Source
+extension ViewController: UICollectionViewDataSource {
+    //Data Source
+    func getMoviesNowPlaying() {
+        let movieRequest = MovieRequest()
+        movieRequest.getMoviesInCinemas { [weak self] result in
+            switch result {
+            case .failure(let error):
+                print(error)
+
+            case .success(let movies):
+                self?.listOfMovies = movies
+                DispatchQueue.main.async {
+                    self?.movieCollectionView.reloadData()
+                }
+            }
+        }
+    }
+    
+    func getMovieGenres(id: Int, indexPath: IndexPath) {
+        let movieRequest = MovieRequest()
+        movieRequest.getMovieGenre(id: id, indexPath: indexPath) { [weak self] result, indexPath in
+            switch result{
+            case .failure(let error):
+                print(error)
+
+            case .success(let genres):
+                self?.genreMap[indexPath] = genres
+                DispatchQueue.main.async {
+                    let cell = self?.movieCollectionView.cellForItem(at: indexPath) as? MovieCollectionViewCell
+                    //Accessing the movie object and set movie title abd rating to it's related labels
+                    cell?.updateMovieGenre(with: genres)
+                }
+            }
+        }
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return listOfMovies.count
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        let movie = listOfMovies[indexPath.row]
+        let id = movie.id
+        getMovieGenres(id: id, indexPath: indexPath)
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        guard let cell = movieCollectionView.dequeueReusableCell(withReuseIdentifier: "MovieCollectionViewCell", for: indexPath) as? MovieCollectionViewCell else {return UICollectionViewCell()}
+        
+        let target = 1
+        //Accessing the movie object and set movie title abd rating to it's related labels
+        if target >= listOfMovies.startIndex && target < listOfMovies.endIndex {
+            let movie = listOfMovies[indexPath.row]
+            cell.updateMovieTitleAndRating(with: movie.title, with: movie.voteAverage)
```

Comment:
At this call site, check if the `with` used in your methods are providing the meaning that you hoped for